### PR TITLE
Use thread-safe ntoa functions

### PR DESCRIPTION
**Pull request by PetervdPerk-NXP**
## Summary
Depends on apache/incubator-nuttx#6751

## Impact
Replaces ntoa by ntoa_r

## Testing
See apache/incubator-nuttx#6751

### CODE REVIEW

File: examples/ipcfg/ipcfg_main.c [hunk 1]
```diff
@@ -281,8 +281,9 @@ static void ipcfg_dump_ipv4addr(FAR const char *variable, in_addr_t address)
       {
         address
       };
+      char inetaddr[INET_ADDRSTRLEN + 2];
```
Comment (pkarashchenko) [inline examples/ipcfg/ipcfg_main.c:284]:
```suggestion

      char inetaddr[INET_ADDRSTRLEN + 2];
```

File: netutils/iperf/iperf.c [hunk 2]
```diff
@@ -397,8 +399,9 @@ static int iperf_run_tcp_server(void)
           actual_recv = recv(sockfd, buffer, want_recv, 0);
           if (actual_recv == 0)
             {
+              char inetaddr[INET_ADDRSTRLEN + 2];
```
Comment (pkarashchenko) [inline netutils/iperf/iperf.c:402]:
```suggestion
              char inetaddr[INET_ADDRSTRLEN];
```
Should be fine

**Review by xiaoxiang781216 — APPROVED** [review]

File: examples/bridge/bridge_main.c [hunk 3]
```diff
@@ -165,7 +166,8 @@ static int briget_net1_setup(void)
     }
 
   dhcpc_close(handle);
-  printf("NET1: Assigned IP: %s\n", inet_ntoa(ds.ipaddr));
+  printf("NET1: Assigned IP: %s\n", inet_ntoa_r(ds.ipaddr, inetaddr,
+                                                sizeof(inetaddr)));
```
Comment (pkarashchenko) [inline examples/bridge/bridge_main.c:170]: Optional
```suggestion
  printf("NET1: Assigned IP: %s\n", 
         net_ntoa_r(ds.ipaddr, inetaddr, sizeof(inetaddr)));
```

File: examples/bridge/bridge_main.c [hunk 4]
```diff
@@ -289,7 +292,8 @@ static int briget_net2_setup(void)
...
```
Comment (pkarashchenko) [inline examples/bridge/bridge_main.c:296]: Optional
```suggestion
  printf("NET1: Assigned IP: %s\n",
         inet_ntoa_r(ds.ipaddr, inetaddr, sizeof(inetaddr)));
```

File: examples/ipcfg/ipcfg_main.c [hunk 5]
```diff
@@ -282,7 +282,10 @@ static void ipcfg_dump_ipv4addr(FAR const char *variable, in_addr_t address)
         address
       };
 
-      printf("%s%s\n", variable, inet_ntoa(saddr));
+      char inetaddr[INET_ADDRSTRLEN];
+
+      printf("%s%s\n", variable, inet_ntoa_r(saddr, inetaddr,
+                                             sizeof(inetaddr)));
```
Comment (pkarashchenko) [inline examples/ipcfg/ipcfg_main.c:288]: Optional
```suggestion
      printf("%s%s\n", variable,
             inet_ntoa_r(saddr, inetaddr, sizeof(inetaddr)));
```

File: examples/discover/discover_main.c [hunk 6]
```diff
@@ -138,6 +138,7 @@ int main(int argc, FAR char *argv[])
       struct dhcpc_state ds;
       dhcpc_request(handle, &ds);
       netlib_set_ipv4addr("eth0", &ds.ipaddr);
+      char inetaddr[INET_ADDRSTRLEN];
```
Comment (pkarashchenko) [inline examples/discover/discover_main.c:141]:
```suggestion
      struct dhcpc_state ds;
      char inetaddr[INET_ADDRSTRLEN];
 
      dhcpc_request(handle, &ds);
      netlib_set_ipv4addr("eth0", &ds.ipaddr);
```

File: examples/webserver/webserver_main.c [hunk 7]
```diff
@@ -165,6 +165,7 @@ int main(int argc, FAR char *argv[])
   if (handle)
     {
       struct dhcpc_state ds;
+      char inetaddr[INET_ADDRSTRLEN];
```
Comment (pkarashchenko) [inline examples/webserver/webserver_main.c:168]:
```suggestion
      char inetaddr[INET_ADDRSTRLEN];

```

File: examples/xmlrpc/xmlrpc_main.c [hunk 8]
```diff
@@ -88,8 +87,14 @@
  * Private Data
  ****************************************************************************/
 
-static const char *notimplemented = { "HTTP/1.1 501 Not Implemented\n\n" };
-static const char *separator = { "\015\012\015\012" };
+static const char *notimplemented =
+{
```
Comment (pkarashchenko) [inline examples/xmlrpc/xmlrpc_main.c:91]: Minor, optional: I think we can omit `{}`

File: examples/xmlrpc/xmlrpc_main.c [hunk 9]
```diff
@@ -178,8 +184,15 @@ static void xmlrpc_handler(int fd)
...
-  int ret, len, max = 0, loadlen = -1;
-  char buffer[CONFIG_EXAMPLES_XMLRPC_BUFFERSIZE] = { 0 };
+  int ret;
+  int len;
+  int max = 0;
+  int loadlen = -1;
+  char buffer[CONFIG_EXAMPLES_XMLRPC_BUFFERSIZE] =
+  {
+    0
+  };
```
Comment (pkarashchenko) [inline examples/xmlrpc/xmlrpc_main.c:194]:
```suggestion
    {
      0
    };
```

File: examples/xmlrpc/xmlrpc_main.c [hunk 10]
```diff
@@ -323,15 +335,16 @@ static int xmlrpc_netinit(void)
...
+  /* Get an IP address.  Note: there is no logic here for renewing the
+   * address in this example.  The address should be renewed in
+   * ds.lease_time/2 seconds.
    */
 
   printf("Getting IP address\n");
   if (handle)
     {
       struct dhcpc_state ds;
+      char inetaddr[INET_ADDRSTRLEN];
```
Comment (pkarashchenko) [inline examples/xmlrpc/xmlrpc_main.c:347]: Optional
```suggestion
      char inetaddr[INET_ADDRSTRLEN];

```

File: netutils/iperf/iperf.c [hunk 11]
```diff
@@ -397,8 +399,9 @@ static int iperf_run_tcp_server(void)
           actual_recv = recv(sockfd, buffer, want_recv, 0);
           if (actual_recv == 0)
             {
+              char inetaddr[INET_ADDRSTRLEN];
               printf("closed by the peer: %s,%d\n",
-                     inet_ntoa(remote_addr.sin_addr),
+                     inet_ntoa_r(remote_addr.sin_addr, inetaddr),
```
Comment (pkarashchenko) [inline netutils/iperf/iperf.c:404]: size is missing

File: netutils/iperf/iperf.c [hunk 12]
```diff
@@ -383,7 +383,9 @@ static int iperf_run_tcp_server(void)
         }
       else
         {
-          printf("accept: %s,%d\n", inet_ntoa(remote_addr.sin_addr),
+          char inetaddr[INET_ADDRSTRLEN];
+          printf("accept: %s,%d\n",
+                 inet_ntoa_r(remote_addr.sin_addr, inetaddr, sizeof(inetaddr)),
```
Comment (pkarashchenko) [inline netutils/iperf/iperf.c:388]: please fix

File: netutils/iperf/iperf.c [hunk 13]
```diff
@@ -383,7 +383,9 @@ static int iperf_run_tcp_server(void)
         }
       else
         {
-          printf("accept: %s,%d\n", inet_ntoa(remote_addr.sin_addr),
+          char inetaddr[INET_ADDRSTRLEN];
```
Comment (pkarashchenko) [inline netutils/iperf/iperf.c:386]:
```suggestion
          char inetaddr[INET_ADDRSTRLEN];

```

File: nshlib/nsh_netcmds.c [hunk 14]
```diff
@@ -1131,7 +1133,7 @@ int cmd_arp(FAR struct nsh_vtbl_s *vtbl, int argc, char **argv)
           goto errout_cmdfaild;
         }
 
-      nsh_output(vtbl, "HWaddr: %s\n",  ether_ntoa(&mac));
+      nsh_output(vtbl, "HWaddr: %s\n",  ether_ntoa_r(&mac, hwaddr));
```
Comment (pkarashchenko) [inline nshlib/nsh_netcmds.c:1136]:
```suggestion
      nsh_output(vtbl, "HWaddr: %s\n", ether_ntoa_r(&mac, hwaddr));
```

File: wireless/wapi/src/wapi.c [hunk 15]
```diff
@@ -294,7 +295,7 @@ static int wapi_show_cmd(int sock, int argc, FAR char **argv)
     }
   else
     {
-      printf("  NetMask: %s\n", inet_ntoa(addr));
+      printf("  NetMask: %s\n", inet_ntoa_r(addr, inetaddr));
```
Comment (pkarashchenko) [inline wireless/wapi/src/wapi.c:298]: size is missing

File: wireless/wapi/src/wapi.c [hunk 16]
```diff
@@ -913,6 +914,7 @@ static int wapi_reconnect_cmd(int sock, int argc, FAR char **argv)
 static int wapi_save_config_cmd(int sock, int argc, FAR char **argv)
 {
   char essid[WAPI_ESSID_MAX_SIZE + 1];
+  char bssid[INET_ADDRSTRLEN];
```
Comment (pkarashchenko) [inline wireless/wapi/src/wapi.c:917]: why `bssid` is `INET` and not `ETH`?

File: examples/xmlrpc/xmlrpc_main.c [hunk 17]
```diff
@@ -88,8 +87,14 @@
  * Private Data
  ****************************************************************************/
 
-static const char *notimplemented = { "HTTP/1.1 501 Not Implemented\n\n" };
-static const char *separator = { "\015\012\015\012" };
+static const char *notimplemented =
```
Comment (pkarashchenko) [inline examples/xmlrpc/xmlrpc_main.c:90]: Not related to this PR, but will leave here for history
```suggestion
static const FAR char *notimplemented =
```

File: examples/xmlrpc/xmlrpc_main.c [hunk 18]
```diff
@@ -88,8 +87,14 @@
...
+{
+  "HTTP/1.1 501 Not Implemented\n\n"
+};
+static const char *separator =
```
Comment (pkarashchenko) [inline examples/xmlrpc/xmlrpc_main.c:94]: Not related to this PR, but will leave here for history
```suggestion
static const FAR char *separator =
```

File: examples/xmlrpc/xmlrpc_main.c [hunk 19]
```diff
@@ -133,7 +138,8 @@ static char *xmlrpc_findbody(char *buf)
  *
  ****************************************************************************/
 
-static int xmlrpc_getheader(char *buffer, char *header, char *value, int size)
+static int xmlrpc_getheader(char *buffer, char *header,
+                            char *value, int size)
```
Comment (pkarashchenko) [inline examples/xmlrpc/xmlrpc_main.c:142]: Not related to this PR, but will leave here for history
```suggestion
static int xmlrpc_getheader(FAR char *buffer, FAR char *header,
                            FAR char *value, int size)
```

File: examples/xmlrpc/xmlrpc_main.c [hunk 20]
```diff
@@ -133,7 +138,8 @@ static char *xmlrpc_findbody(char *buf)
...
 {
   char *temp;
```
Comment (pkarashchenko) [inline examples/xmlrpc/xmlrpc_main.c:144]: Not related to this PR, but will leave here for history
```suggestion
  FAR char *temp;
```

File: examples/xmlrpc/xmlrpc_main.c [hunk 21]
```diff
@@ -178,8 +184,15 @@ static void xmlrpc_handler(int fd)
...
+  int len;
+  int max = 0;
+  int loadlen = -1;
+  char buffer[CONFIG_EXAMPLES_XMLRPC_BUFFERSIZE] =
+  {
+    0
+  };
+
   char value[CONFIG_XMLRPC_STRINGSIZE + 1];
   char *temp;
```
Comment (pkarashchenko) [inline examples/xmlrpc/xmlrpc_main.c:197]: Not related to this PR, but will leave here for history
```suggestion
  FAR char *temp;
```

File: examples/bridge/bridge_main.c [hunk 22]
```diff
@@ -165,7 +166,8 @@ static int briget_net1_setup(void)
     }
 
   dhcpc_close(handle);
-  printf("NET1: Assigned IP: %s\n", inet_ntoa(ds.ipaddr));
+  printf("NET1: Assigned IP: %s\n",
```
Comment (pkarashchenko) [inline examples/bridge/bridge_main.c:169]:
```suggestion
  printf("NET1: Assigned IP: %s\n",
```

File: examples/tcpecho/tcpecho_main.c [hunk 23]
```diff
@@ -243,12 +244,14 @@ static int tcpecho_server(void)
...
         {
+          char inetaddr[INET_ADDRSTRLEN];
+
           /* new client connection */
 
           clilen = sizeof(cliaddr);
           connfd = accept(listenfd, (struct sockaddr *)&cliaddr, &clilen);
 
-          ninfo("new client: %s\n", inet_ntoa(cliaddr.sin_addr));
+          ninfo("new client: %s\n", inet_ntoa_r(cliaddr.sin_addr, inetaddr));
```
Comment (pkarashchenko) [inline examples/tcpecho/tcpecho_main.c:254]:
```suggestion
          ninfo("new client: %s\n",
                inet_ntoa_r(cliaddr.sin_addr, inetaddr, sizeof(inetaddr)));
```

**Comment by pkarashchenko** [comment]
Please squash commits

File: wireless/wapi/src/wapi.c (hunk 16 above)
Comment (xiaoxiang781216) [reply wireless/wapi/src/wapi.c]: WiFi use the same MAC as Ethernet

**Comment by xiaoxiang781216** [comment]
@PetervdPerk-NXP could you fix the remaining warning as much as possible?

File: wireless/wapi/src/wapi.c (hunk 16 above)
Comment (pkarashchenko) [reply wireless/wapi/src/wapi.c]: Yes. `INET_ADDRSTRLEN` is for IP address, so it is changed to 20.

File: examples/discover/discover_main.c [hunk 24]
```diff
@@ -136,6 +136,8 @@ int main(int argc, FAR char *argv[])
   if (handle)
     {
       struct dhcpc_state ds;
+      char inetaddr[INET_ADDRSTRLEN];
+
```
Comment (pkarashchenko) [inline examples/discover/discover_main.c:140]:
```suggestion

```

**Comment by xiaoxiang781216** [comment]
@PetervdPerk-NXP there are some errors, please fix it, thanks.

File: examples/tcpecho/tcpecho_main.c [hunk 25]
```diff
@@ -94,6 +94,7 @@ static int tcpecho_netsetup()
   struct dhcpc_state ds;
   void *handle;
 #endif
+  char inetaddr[INET_ADDRSTRLEN];
```
Comment (pkarashchenko) [inline examples/tcpecho/tcpecho_main.c:97]:
```suggestion
  char inetaddr[INET_ADDRSTRLEN];
#endif
```

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**